include/etcpal/log.h

```diff
@@ -0,0 +1,407 @@
+ * SP:                         1
+ * PROCID:          [Referenced]
+ * SP:                         1
+ * MSGID (not used):           1
+ * SP:                         1
+ * STRUCTURED-DATA (not used): 1
+ * SP:                         1
+ * -----------------------------
+ * Total non-referenced:      17
+ */
+/*! The maximum length of the syslog header. */
+#define ETCPAL_SYSLOG_HEADER_MAX_LEN                                                                                 \
+  (17u + (ETCPAL_LOG_TIMESTAMP_LEN - 1u) + (ETCPAL_LOG_HOSTNAME_MAX_LEN - 1u) + (ETCPAL_LOG_APP_NAME_MAX_LEN - 1u) + \
+   (ETCPAL_LOG_PROCID_MAX_LEN - 1u))
+
+/*! The minimum length of a buffer passed to etcpal_create_syslog_str(). */
+#define ETCPAL_SYSLOG_STR_MIN_LEN ETCPAL_SYSLOG_HEADER_MAX_LEN
+/*! The minimum length of a buffer passed to etcpal_create_log_str(). */
+#define ETCPAL_LOG_STR_MIN_LEN (ETCPAL_LOG_TIMESTAMP_LEN + 1u /*SP*/ + 6u /*pri*/ + 1u /*SP*/)
+
+/*! The maximum length of a syslog string that will be passed to an etcpal_log_callback function. */
+#define ETCPAL_SYSLOG_STR_MAX_LEN (ETCPAL_SYSLOG_HEADER_MAX_LEN + ETCPAL_RAW_LOG_MSG_MAX_LEN)
+
+/* Human-reaadable log string max length:
+ * Timestamp:      [Referenced]
+ * Space:                     1
+ * Priority:                  6 ([CRIT])
+ * Space:                     1
+ * Message length: [Referenced]
+ * ----------------------------
+ * Total non-referenced:      8
+ */
+/*! The maximum length of a string that will be passed via the human_readable member of an
+ *  EtcPalLogStrings struct. */
+#define ETCPAL_LOG_STR_MAX_LEN (8u + (ETCPAL_LOG_TIMESTAMP_LEN - 1u) + ETCPAL_RAW_LOG_MSG_MAX_LEN)
+
+/*! A set of parameters which represent the current local time with millisecond resolution. */
+typedef struct EtcPalLogTimeParams
+{
+  int year;       /*!< Absolute year. Valid range 0-9999. */
+  int month;      /*!< Month of the year. Valid range 1-12 (starting with 1 for January). */
+  int day;        /*!< Day of the month. Valid range 1-31.  */
+  int hour;       /*!< Hours since midnight. Valid range 0-23. */
+  int minute;     /*!< Minutes past the current hour. Valid range 0-59. */
+  int second;     /*!< Seconds past the current minute. Valid range 0-60 (to handle leap seconds). */
+  int msec;       /*!< Milliseconds past the current second. Valid range 0-999. */
+  int utc_offset; /*!< The local offset from UTC in minutes. */
+} EtcPalLogTimeParams;
+
+/*! The set of log strings passed with a call to an etcpal_log_callback function. Any members not
+ *  requested in the corresponding EtcPalLogParams struct will be NULL. */
+typedef struct EtcPalLogStrings
+{
+  /*! Log string formatted compliant to RFC 5424. */
+  const char* syslog;
+  /*! Log string formatted for readability per ETC convention. */
+  const char* human_readable;
+  /*! The original log string that was passed to etcpal_log() or etcpal_vlog(). Will overlap with
+   *  one of syslog_str or human_str. */
+  const char* raw;
```

Comment:
We should include the original log priority as an integer here so that you can forward log messages to the syslog daemon by passing the `raw` string and the priority.